synapseclient/client.py

```diff
@@ -626,6 +626,27 @@ def is_certified(self, user: typing.Union[str, int]) -> bool:
                 return False
             raise
 
+    def is_synapse_id(self, syn_id):
+        """Checks if given synID is valid (attached to actual entity?)"""
+        if isinstance(syn_id, str):
```

Comment:
I forget but does `syn.get` first use the `utils.is_synapse_id_str`?

Reply:
It does, but passing any list-like collections will raise a TypeError.  I added this type check to handle those situations.